From: Catalin Marinas <catalin.marinas@arm.com>
Subject: kmemleak-avoid-rcu-stalls-when-freeing-metadata-for-per-cpu-pointers-v2
Date: Wed, 6 Dec 2023 11:44:14 +0000

update the comment to include the *_percpu() callbacks, not just the *_phys()

Link: https://lkml.kernel.org/r/20231206114414.2085824-1-catalin.marinas@arm.com
Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>
Reported-by: Waiman Long <longman@redhat.com>
Reviewed-by: Waiman Long <longman@redhat.com>
Link: https://lore.kernel.org/r/20231127194153.289626-1-longman@redhat.com
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/kmemleak.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/mm/kmemleak.c~kmemleak-avoid-rcu-stalls-when-freeing-metadata-for-per-cpu-pointers-v2
+++ a/mm/kmemleak.c
@@ -21,8 +21,8 @@
  *   red black trees used to look-up metadata based on a pointer to the
  *   corresponding memory block. The kmemleak_object structures are added to
  *   the object_list and the object tree root in the create_object() function
- *   called from the kmemleak_alloc() (or kmemleak_alloc_phys()) callback and
- *   removed in delete_object() called from the kmemleak_free() callback
+ *   called from the kmemleak_alloc{,_phys,_percpu}() callback and removed in
+ *   delete_object() called from the kmemleak_free{,_phys,_percpu}() callback
  * - kmemleak_object.lock (raw_spinlock_t): protects a kmemleak_object.
  *   Accesses to the metadata (e.g. count) are protected by this lock. Note
  *   that some members of this structure may be protected by other means
_
